main.js

```diff
@@ -0,0 +1,310 @@
+var weather = []
+
+//initialize an object to contain current conditions
+var today = {}
+
+//array of days of the week, used with Moment.js to convert day number into day name string
+var daysofWeek = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+
+//sets defaultLocationName to an empty string, will populate later if there is a default location
+var defaultLocationName = ""
+
+//if a default location has been set, load that data automatically
+var storedData = localStorage.getItem("location")
+if (storedData) {
+  var defaultLocation = JSON.parse(storedData)
+  //show a spinner while fetching data from the API
+  $('.weekly-forecast').addClass('hide')
+  $('.daily-forecast').addClass('hide')
+  $('.spinner').removeClass('hide')
+  //pass location query and true for isDefault because we're defining this as the default location
+  getLocationCoordinates(defaultLocation, true)
+}
+
+//use geolocation data to find locality name
+$('.locate-me').on('click', function() {
+  if ("geolocation" in navigator) {
+    //show a spinner while fetching data from the API
+    $('.weekly-forecast').addClass('hide')
+    $('.daily-forecast').addClass('hide')
+    $('.spinner').removeClass('hide')
+    navigator.geolocation.getCurrentPosition(function(position) {
+      locationLatitude = position.coords.latitude
+      locationLongitude = position.coords.longitude
+      getLocationAddress(locationLatitude, locationLongitude)
+    });
+  } else {
+    alert('location services not available')
+  }
+})
+
+//on search get location information and pass it to the weather APIs
+$('.search').on('click', function() {
+  input = $('#search-query').val();
+  //show a spinner while fetching data from the API
+  $('.weekly-forecast').addClass('hide')
+  $('.daily-forecast').addClass('hide')
+  $('.spinner').removeClass('hide')
+  //pass location query and false because we are not defining this as the default location
+  getLocationCoordinates(input, false)
+})
+
+//set a default location and save it in localStorage, by passing this location to getLocationCoordinates with boolean true we set it as the default location
+$('body').on("click", ".default-location", function() {
+  var input = $('#search-query').val();
+  localStorage.setItem('location', JSON.stringify(input))
+  getLocationCoordinates(input, true)
+})
+
+//used in geolocation: gets a location name from the location address data to pass to the getLocationCoordinates function to get weather data
+var parseLocation = function(data) {
+  //iterates through location address data to find the item that gives the locality or sublocality name to pass to the getLocationCoordinates function. If no name, alerts an error and hides the spinner
+  var locationName
+  for (i = 0; i < data.results.length; i++) {
+    for (j = 0; j < data.results[i].types.length; j++) {
+      if (data.results[i].types[j] == "locality" || data.results[i].types[j] == "sublocality") {
```

Comment:
Why is this the check?  there are more specific types of results to use (https://developers.google.com/maps/documentation/geocoding/intro#Types), couldn't you have gotten the result from something like street_address type and then found the city/state combo?  Think of a big locality (like NYC), the weather might be different in the different boroughs, being as specific as possible would be ideal here.  Also, when users say "use my location" then want something very tailored to them, not generic.